### ISSUE TITLE
[BUG] npm uninstalls itself when installing other packages (under nvm for Windows)

### ISSUE THREAD

**Issue by davidgeary** [npm/cli#2593]
### Current Behavior:
I'm using [nvm on Windows](https://github.com/coreybutler/nvm-windows/) to manage npm versions and have just installed npm 7.4.3 via node 15.7.0.  If I then install a package, e.g. typescript, I see:

```sh
added 1 package, removed 252 packages, and audited 2 packages in 4s

found 0 vulnerabilities
```

Note the 252 packages removed. After that, the _npm_ folder under _%APPDATA%\nvm\v15.7.0\node_modules_ has been deleted and naturally, all npm CLI commands then fail.

This works correctly with npm 6.9.0/node 12.3.1 and npm 6.14.10/node 14.15.4 (with all three environments installed using NVM4W 1.1.7, so that's not changed).

I'm aware that NVM4W uses symlinks to control the active version of node/npm, so can the use of symlinks be causing npm 7.x to uninstall itself when installing another package?

### Expected Behavior:
npm should still be installed and working after installing other packages.

### Steps To Reproduce:
1. Download and install [NVM4W 1.1.7](https://github.com/coreybutler/nvm-windows/releases/tag/1.1.7) (on Windows obviously)
2. Open a command window as administrator and run the following commands:
    1. `nvm install 15.7.0` - this should install node 15.7.0 and npm 7.4.3
    2. `nvm use 15.7.0` - this sets 15.7.0/7.4.3 as the active versions
    3. `cd %APPDATA%\nvm\v15.7.0\node_modules`
    4. `dir`
    This will list the folders for all installed packages; you should see _npm_.
    5. `npm install -g typescript`
    6. `dir`
    Again, this will list folders for the installed packages, which should now be _npm_ and _typescript_, but is actually _.bin_ and _typescript_

### Environment:
- OS: Windows 10 Pro (10.0.18363)
- Node: 15.7.0
- npm: 7.4.3
- NVM4W: 1.1.7

**Comment by aminomancer** [comment]
I can reproduce, also on 18363. And npm 6.x works without a hitch for me too.

**Comment by darcyclarke** [comment]
@davidgeary can you try to reproduce in the latest v7? (ie. `npm i -g npm`)

**Comment by davidgeary** [comment]
@darcyclarke Ah, it's working properly with 15.8.0/7.5.1 - thanks for the prompt to retest.

Turns out that a new version of node was released the day after I found the issue.  That version - 15.8.0 - now includes npm 7.5.1, which is working fine.

FYI, I tried upgrading it to the very latest (7.5.2) using `npm i -g npm`, but it just blew up with an `EPERM: operation not permitted` error while renaming the folders, and corrupted the npm install.  And _that's_ why I use NVM4W rather than trying to update npm directly 😉
